packages/sync-actions/src/utils/common-actions.js

```diff
@@ -19,6 +22,7 @@ export const createIsEmptyValue = (emptyValues) => (value) =>
  * @param  {Object} options.newObj - the new representation of the object
  * @param {Boolean} options.shouldOmitEmptyString - a flag to determine if we should treat an empty string a NON-value
  * @param {Boolean} options.shouldUnsetOmittedProperties - a flag to determine if we should unset fields which are omitted in the newObj
+ * @param {Boolean} options.shouldPreventUnsettingRequiredFields - a flag to determine if required fields should be unset
```

Comment:
somehow that flag helps to generate invalid actions, `{ action: 'changeName' },` 🤔.

maybe to make the package backward compatible, change this flag to be warper for the new added functionality.
so in case the user wanted new behaviour, the `xyz` flag set to true, otherwise the old behaviour works, which will enable us to pump the version as `patch`, wdyt?!

hope that i'm not trolling you :)